src/main/java/com/exadel/ipromise/controllers/UserController.java

```diff
@@ -0,0 +1,38 @@
+package com.exadel.ipromise.controllers;
+
+import com.exadel.ipromise.dto.UserDto;
+import com.exadel.ipromise.service.UserService;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpSession;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:8080")
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/registration")
+    public UserDto create(@RequestBody UserDto userDto, HttpSession session) {
+        return userService.addUser(userDto, session);
+    }
+
+    @PostMapping("/login")
+    public UserDto logIn(@RequestBody UserDto userDto, HttpSession session) {
+        return userService.logIn(userDto, session);
+    }
+
+    @GetMapping("/logout")
+    public
+    ResponseEntity<?> logout(HttpSession session) {
```

Comment:
don't use wild card, If you return String - return String

Reply:
Done